### PR TITLE
feat: Integration testing

### PR DESCRIPTION
**Pull request by antoinegelloz**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -120,3 +120,31 @@ jobs:
...
+    - Tests
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+    - uses: actions/setup-go@v3
```
Comment (antoinegelloz) [inline .github/workflows/main.yml:136]: Can you also add:
```
        with:
          go-version-file: ./components/${{ matrix.folder }}/go.mod
          cache-dependency-path: ./components/${{ matrix.folder }}/go.sum
          cache: true
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -120,3 +120,31 @@ jobs:
           build-args: |
             APP_SHA=${{ github.sha }}
             VERSION=main
```
Comment (antoinegelloz) [inline .github/workflows/main.yml:122]: Would it be a good time to change it?
```suggestion
            VERSION=${{ github.sha }}
```

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -104,6 +108,33 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - uses: actions/setup-go@v3
```
Comment (antoinegelloz) [inline .github/workflows/pull_request.yml:122]: Also here:
`        with:
          go-version-file: ./components/${{ matrix.folder }}/go.mod
          cache-dependency-path: ./components/${{ matrix.folder }}/go.sum
          cache: true
`

File: Taskfile.yaml [hunk 4]
```diff
@@ -36,6 +41,7 @@ tasks:
       - task wallets:tests
       - task webhooks:tests
       - task ledger:tests
+      - task machine:tests
```
Comment (antoinegelloz) [inline Taskfile.yaml:44]: to remove

File: Taskfile.yaml [hunk 5]
```diff
@@ -48,6 +54,7 @@ tasks:
       - task wallets:lint
       - task webhooks:lint
       - task ledger:lint
+      - task machine:lint
```
Comment (antoinegelloz) [inline Taskfile.yaml:57]: to remove

File: Taskfile.yaml (hunk 4 above)
Comment (antoinegelloz) [reply Taskfile.yaml:44]: and maybe add
`- task integration:tests` ?

**Review by gfyrag — APPROVED** [review]

**Review by paul-nicolas — APPROVED** [review]

**Pull request merged**